deepchem/dock/docking.py

```diff
@@ -1,119 +1,96 @@
-    pdbbind_tasks = ["-logKd/Ki"]
-    n_features = 2052
-    model = MultitaskRegressor(
-        len(pdbbind_tasks),
-        n_features,
-        dropouts=[.25],
-        learning_rate=0.0003,
-        weight_init_stddevs=[.1],
-        batch_size=64,
-        model_dir=self.model_dir)
-    model.reload()
-
-    self.pose_scorer = GridPoseScorer(model, feat="grid")
-    self.pose_generator = VinaPoseGenerator(
-        exhaustiveness=exhaustiveness, detect_pockets=detect_pockets)
+    self.pose_generator = pose_generator
+    self.featurizer = featurizer
+    self.scoring_model = scoring_model
 
   def dock(self,
-           protein_file,
-           ligand_file,
+           molecular_complex,
            centroid=None,
            box_dims=None,
-           dry_run=False):
-    """Docks using Vina and DNNs."""
-    protein_docked, ligand_docked = self.pose_generator.generate_poses(
-        protein_file, ligand_file, centroid, box_dims, dry_run)
-    if not dry_run:
-      score = self.pose_scorer.score(protein_docked, ligand_docked)
+           exhaustiveness=10,
+           num_modes=9,
+           num_pockets=None,
+           out_dir=None):
+    """Docks using Vina and RF.
+
+    Parameters
+    ----------
+    molecular_complex: Object
+      Some representation of a molecular complex.
+    exhaustiveness: int, optional (default 10)
+      Tells Autodock Vina how exhaustive it should be with pose
+      generation.
+    num_modes: int, optional (default 9)
+      Tells Autodock Vina how many binding modes it should generate at
+      each invocation.
+    num_pockets: int, optional (default None)
+      If specified, `self.pocket_finder` must be set. Will only
+      generate poses for the first `num_pockets` returned by
+      `self.pocket_finder`.
+    out_dir: str, optional (default None)
+      If specified, write generated poses to this directory.
+    use_pose_generator_scores: bool, optional (default False)
+      If `True`, ask pose generator to generate scores. This cannot be
+      `True` if `self.featurizer` and `self.scoring_model` are set
+      since those will be used to generate scores in that case. 
+    """
+    outputs = self.pose_generator.generate_poses(
+        molecular_complex,
+        centroid=centroid,
+        box_dims=box_dims,
+        exhaustiveness=exhaustiveness,
+        num_modes=num_modes,
+        num_pockets=num_pockets,
+        out_dir=out_dir,
+        generate_scores=use_pose_generator_scores)
+    if use_pose_generator_scores:
+      complexes, scores = outputs
     else:
-      score = np.zeros((1,))
-    return (score, (protein_docked, ligand_docked))
-'''
+      complexes = outputs
+    for posed_complex in complexes:
+      if self.featurizer is not None:
+        # TODO: How to handle the failure here?
+        features, _ = self.featurizer.featurize_complexes([molecular_complex])
+        dataset = NumpyDataset(X=features)
+        score = self.model.predict(dataset)
```

Comment:
Should this be `self.scoring_model`?

Reply:
Good catch! This code turned out to be buggy. I've added an extra unit test for this path to make sure it's run in the tests.